contrib/platform/src/com/sun/jna/platform/win32/Winioctl.java

```diff
@@ -72,4 +95,320 @@ public STORAGE_DEVICE_NUMBER(Pointer memory) {
+
+        /**
+         * Specifies the byte offset from the beginning of the structure to a null-terminated ASCII string that contains
+         * the device's vendor ID. If the device has no vendor ID, this member is zero.
+         */
+        public WinDef.DWORD vendorIdOffset;
+
+        /**
+         * Specifies the byte offset from the beginning of the structure to a null-terminated ASCII string that contains
+         * the device's product ID. If the device has no product ID, this member is zero.
+         */
+        public WinDef.DWORD productIdOffset;
+
+        /**
+         * Specifies the byte offset from the beginning of the structure to a null-terminated ASCII string that contains
+         * the device's product revision string. If the device has no product revision string, this member is zero.
+         */
+        public WinDef.DWORD productRevisionOffset;
+
+        /**
+         * Specifies the byte offset from the beginning of the structure to a null-terminated ASCII string that contains
+         * the device's serial number. If the device has no serial number, this member is zero.
+         */
+        public WinDef.DWORD serialNumberOffset;
+
+        /**
+         * Specifies an enumerator value of type STORAGE_BUS_TYPE that indicates the type of bus to which the device is
+         * connected. This should be used to interpret the raw device properties at the end of this structure (if any).
+         */
+        public /*STORAGE_BUS_TYPE*/ WinDef.UCHAR busType;
+
+        /**
+         * Indicates the number of bytes of bus-specific data that have been appended to this descriptor.
+         */
+        public WinDef.DWORD rawPropertiesLength;
+
+        /**
+         * Contains an array of length one that serves as a place holder for the first byte of the bus specific property
+         * data.
+         */
+        public byte[] rawDeviceProperties = new byte[1];
+
+        public static class ByReference extends STORAGE_DEVICE_DESCRIPTOR implements Structure.ByReference {
+            public ByReference() {
+            }
+
+            public ByReference(Pointer memory) {
+                super(memory);
+            }
+        }
+
+        public STORAGE_DEVICE_DESCRIPTOR() {
+        }
+
+        public STORAGE_DEVICE_DESCRIPTOR(Pointer memory) {
+            super(memory);
+            read();
+
+            // Allocate the buffer for the raw properties, and re-read
+            rawDeviceProperties = new byte[rawPropertiesLength.intValue()];
+            read();
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return FIELDS;
+        }
+    }
+
+    /**
+     * Describes the extended geometry of disk devices and media.
+     */
+    class DISK_GEOMETRY_EX extends Structure {
+        public static class ByReference extends DISK_GEOMETRY_EX implements Structure.ByReference {
+            public ByReference() {
+            }
+
```

Comment:
Don't leave this empty - add call to (implicit) _super()_ as this enables setting a breakpoint in the constructor